coord/src/module.c

```diff
@@ -1215,17 +1649,16 @@ int FlatSearchCommandHandler(RedisModuleBlockedClient *bc, RedisModuleString **a
   // adding the WITHSCORES option anyway immediately after the query.
   // Worst case it will appears twice.
   MRCommand_AppendArgsAtPos(&cmd, 3 + req->profileArgs, 1, "WITHSCORES");
-  if (req->withSortby) {
-    // if sort by requested we adding the WITHSORTKEYS option anyway immediately after the query.
-    // Worst case it will appears twice.
-    MRCommand_AppendArgsAtPos(&cmd, 3 + req->profileArgs, 1, "WITHSORTKEYS");
-    // req->withSortingKeys = 1;
+
+  if(req->specialCases) {
+    sendRequiredFields(req, &cmd);
   }
 
+
   struct MRCtx *mrctx = MR_CreateCtx(ctx, req);
   // we prefer the next level to be local - we will only approach nodes on our own shard
   // we also ask only masters to serve the request, to avoid duplications by random
-  MR_SetCoordinationStrategy(mrctx, MRCluster_FlatCoordination | MRCluster_MastersOnly);
```

Comment:
Why was this removed? IIRC this fixes a bug we just had of duplicate results. If its a wrong merge we need to make sure the other conflicts was not resolved incorrectly ...

Reply:
by accident

Reply:
Did you check other merge conflict that did not removed important fixes accidentally?